index.js

```diff
@@ -1,29 +1,48 @@
 const passport     = require('passport');
 const ldapStrategy = require('passport-ldapauth');
+const allowedAttributes = [
+  'dn', 'controls', 'objectClass', 'ou', 'o', 'ibmSerialNumber', 'employeeCountryCode', 
+  'isManager', 'managerCountryCode', 'co', 'uid', 'c', 'serialNumber', 'hrActive', 
+  'hrCompanyCode', 'hrCountryCode', 'hrFirstName', 'hrManagerIndicator', 'emailAddress', 
+  'mail', 'notesEmail', 'notesId', 'notesMailDomain', 'notesShortName', 'primaryNode', 'primaryUserId', 
+  'infotelephonenumber', 'personaltitle', 'jobresponsibilities', 'hrEmployeeType', 'employeeType', 
+  'preferredIdentity', 'iptelephonenumber', 'telephoneNumber', 'tieline', 'alternateTelephoneNumber', 
+  'mobile', 'timestampBpgui', 'hrLastName', 'coreDataIntegrity', 'directoryAlias', 'odmEntry', 'div', 
+  'notesMailFile', 'notesMailServer', 'buildingName', 'floor', 'ibmLoc', 'physicalDeliveryOfficeName', 
+  'workLoc', 'workLocation', 'workplaceIndicator', 'givenName', 'sn', 'timestampFeed', 'costCenter', 
+  'dept', 'divDept', 'hrOrganizationCode', 'callupName', 'cn', 'entryType', 'hrOrgUnitCode', 'manager', 
+  'managerSerialNumber', 'timestampHRMS', 'preferredLastName', 'timestampBpguiRI', 'passwordIsExpired', 
+  'passwordisreset', 'passwordIsResetByAdmin', 'passwordIsStruckOut', 'passwordModifyTimestamp']
 
 // Setup passport with IBM's LDAP config
 const config = { server: {
     url: 'ldaps://bluepages.ibm.com:636',
     searchBase: 'ou=bluepages,o=ibm.com',
-    searchFilter: '(mail={{username}})',
-    searchAttributes: ['cn', 'uid', 'mail']
```

Comment:
I think you referring to `searchFilter` right? if so, well right now the LDAP is used to search based on the users email... the `{{username}}` is a mere reference to the field from passport, what matter here is the `mail=` I think we could also change to `uid=` and authenticating using the uid as well... 
but to be honest with you this could be a whole new feature because we can also implement a authenticate way to identify when is email or uid and the module will allow the authentication with both.... to be honest, not a big deal something like

```javascript
if (username.indexOf("@") >= 0) {
 //mail=
} else {
 //uid=
}
```

but again, this isn't the objective of this PR, this one is just to allow customize the attributes.